sequencer/src/hotshot_commitment.rs

```diff
@@ -106,7 +119,12 @@ async fn sequence(hard_block_limit: usize, hotshot: HotShotClient, contract: Hot
             sleep(RETRY_DELAY).await;
         } else {
             // If we succeed, increase the limit
-            soft_block_limit = std::cmp::min(soft_block_limit * 2, hard_block_limit)
+            soft_block_limit = std::cmp::min(soft_block_limit * 2, hard_block_limit);
+            // Create an exponential distribution for sampling delay times. The distribution should have
+            // mean `delay`, or parameter `\lambda = 1 / delay`.
+            let delay_distr = rand_distr::Exp::<f64>::new(1f64 / delay.as_millis() as f64).unwrap();
+            let delay = Duration::from_millis(delay_distr.sample(&mut rng) as u64);
+            sleep(delay).await;
```

Comment:
What does this do if `delay` is 0, `1f64 / 0f64` is infinity? Maybe better to make `delay` an `Option<Duration>` and skip the whole distribution/delay if it is `None`

Reply:
Good idea, made this optional